### PR TITLE
chore: update OpenTelemetry Collector to 0.94.0-sumo-2

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
chore: update OpenTelemetry Collector to 0.94.0-sumo-2

### Checklist

- [x] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [x] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: tests/integration/internal/constants.go [hunk 1]
```diff
@@ -228,6 +228,13 @@ var (
 		"otelcol_otelsvc_k8s_service_table_size",
 		"otelcol_otelsvc_k8s_owner_table_size",
 		"otelcol_exporter_send_failed_metric_points",
+		"otelcol_http_client_request_size",
+		"otelcol_http_client_response_size",
+		"otelcol_http_client_duration_bucket",
+		"otelcol_http_client_duration_sum",
+		"otelcol_http_server_response_size",
+		"otelcol_http_client_duration_count",
+		"otelcol_http_server_request_size",
```
Comment (swiatekm-sumo) [inline tests/integration/internal/constants.go:237]: What are these metrics?

Comment (kkujawa-sumo) [reply tests/integration/internal/constants.go:237]: It looks like otelcol exposes more metrics by default but there are also other problems, I'll take a look on this closer on Monday.

**Comment by swiatekm-sumo** [comment]
Metrics E2E tests fail because we have some new metrics. We should probably drop the buckets and keep the rest. Maybe it's fine to keep them all? Depends on how many timeseries this is in practice per Pod.

Logs E2E tests fail due to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30797 changing the behaviour of the recombine operator. This is more serious and can't easily be worked around.

**Comment by kkujawa-sumo** [comment]
Note about new metrics exposed by otelcol from tests:
```
Test_Helm_Default_OT_NamespaceOverride/metrics/expected_metrics_are_present 2024-03-11T09:57:43Z retry.go:103
WaitUntilExpectedMetricsPresent() returned an error: found the following unexpected metrics: 
[otelcol_http_client_response_size
 otelcol_http_client_duration_count 
 otelcol_http_server_response_size 
 otelcol_http_client_request_size 
 otelcol_http_client_duration_bucket 
 otelcol_http_server_request_size 
 otelcol_http_client_duration_sum]. 
Sleeping for 3s and will try again.

```

**Comment by kkujawa-sumo** [comment]
`otelcol_http_client*` metrics are exposed by logs and metrics collectors, in default configuration I see additional 20 times series for these metrics:
- otelcol_http_client_response_size
- otelcol_http_client_request_size 
- otelcol_http_client_duration_count 
- otelcol_http_client_duration_bucket
- otelcol_http_client_duration_sum

Screenshot 2024-03-11 at 15 16 52

Screenshot 2024-03-11 at 15 17 12

Screenshot 2024-03-11 at 15 18 31

new `otelcol_http_server*` metrics are exposed by otelcol logs and otelcol metrics
- otelcol_http_server_request_size 
- otelcol_http_server_response_size


Screenshot 2024-03-11 at 15 31 53





@swiatekm-sumo in default configuration `otelcol_http_server_duration_bucket`  metric is collected ([accepted in tests ](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/ce267f5487a918d189c8f56c513c764715e74bdd/tests/integration/internal/constants.go#L303)) so why not collect `otelcol_http_client_duration_bucket`? 🤔

**Pull request closed without merging**